examples/assume_role.py

```diff
@@ -60,7 +58,7 @@
 """
 
 credentials_provider = AssumeRoleProvider(
-    client, Policy=restricted_upload_policy)
+    get_assume_role_creds=lambda: client._get_assume_role_creds(policy=restricted_upload_policy))
```

Comment:
It is bad idea to use private method. You can make `get_assume_role_creds` public.